43 constrain user data

thegooders opened the pull request.

gbhachu committed:
Some CSS style added

thesedatedprince committed:
changed link font

gbhachu committed:
CSS added footer

gbhachu committed:
footer and fonts changed

thesedatedprince committed:
Styled forms

gbhachu committed:
Welcome message and line spacing amended

thegooders committed:
diplay income by user id

thesedatedprince committed:
Updated forms, removed alert message

gbhachu committed:
CSS'd the graphs a bit

thegooders committed:
projections and expend by id

thesedatedprince committed:
Started tables for data output

thegooders committed:
all tests passing, branch complete

gbhachu committed:
added expenditures table

gbhachu committed:
merge conflicts fixed

olwend commented:
21 tests green. 
Routes checked via rake routes.
 ```bin/rake db:rese```t and ```bin/rails s```  successful
See dashboard with no data and errors across base 
"No Incomes Added No Expenditures Added No Projections Added No goals yet".
Data is in the DB, the chart shows blank - messages visible from controllers "No Incomes" etc... is that expected for current build?

@olwen, I think unless you're logging in as the seeded user (john@gmail.com/topsecret), the chart will be blank/should be blank

olwend commented:
Whilst running model tests - we are waiting for user projections association.  A migration for it is not in this build, but the relationships are in user.rb and projection.rb
run ```$ bin/rails g migration AddUserRefToProjections user:references```

Putterhead committed:
Update home.html.erb

Just change the welcome to 'Welcome to Millionaire Machine', which was failing on ./spec/features/static_home_spec.rb:6

olwend commented:
:shipit:
